### ISSUE TITLE
Is it able to change the behavior of  to_string for `Money.Ecto.Composite.Type` ?

### ISSUE THREAD

**Issue by Daniel-Xu** [kipcole9/money_sql#23]
Right now, if we use `en` as default locale, the to_string return `$132` for currency.
However, this is not recognized by DOM field. Is there an option to change this behavior? 

![image](https://user-images.githubusercontent.com/548144/158434460-df3e383a-2e1e-4ece-b84d-98f89c1bedda.png)

**Comment by kipcole9** [comment]
Daniel, thanks for the report. Can you send me the HTML of the relevant part? What are you trying to achieve? What output are you looking for?

**Comment by kipcole9** [comment]
BTW, the actual number formatting is done by `Cldr.Number.to_string/3` so you can check the options available there too.

**Comment by Daniel-Xu** [comment]
I'm using live view and live validation, the html is normal `number_input`
```
                <%= label f, :single_price, "总价格", class: "uppercase md:text-sm text-xs text-gray-500 text-light font-semibold" %>
                <%= number_input f, :single_price, autocomplete: "off">
```

**Comment by Daniel-Xu** [comment]
From the error in console, I'm guessing the to_string result of `Money.Ecto.Composite.Type` can't work well with live_view dom diff. 

What I want to achieve is to be able to change the output of `to_string` for `Money.Ecto.Composite.Type` to `333` instead of `$333`. Hope it's clear.

**Comment by Daniel-Xu** [comment]
I was able to solve this by: 

1. change `number_input` into `text_input`
2. using `currency_symbol: ""` (this is not necessary, but it makes the field look like a number field)

**Comment by kipcole9** [comment]
Daniel, the error is probably because you are using a number input field and providing it with a currency string that includes non-digit input.  

This has nothing to do with `Money.Ecto.Composite.Type`.  The only relevant type is [Money.t](https://hexdocs.pm/ex_money/Money.html#t:t/0) which combines a currency code and a decimal amount.

I believe you have a couple of choices:

1. Don't use a [number input](https://hexdocs.pm/phoenix_html/Phoenix.HTML.Form.html#number_input/3) type, use a [text input](https://hexdocs.pm/phoenix_html/Phoenix.HTML.Form.html#text_input/3) type. And then use [Money.parse/2](https://hexdocs.pm/ex_money/Money.html#parse/2) in your controller to parse it back into a `Money.t`. This allows you quite a flexible input mechanism but it does mean you don't get any benefit from browser-side number formatting and validation. You do get a "round trip" experience though in which the output string can be parsed back into a `Money.t` if you need.

2. Just format the currency amount as a number and pass the number into the DOM. You can get the amount with [Money.to_decimal/1](https://hexdocs.pm/ex_money/Money.html#to_decimal/1) and then format it any way you want ([Cldr.Number.to_string/3](https://hexdocs.pm/ex_cldr_numbers/Cldr.Number.html#to_string/3) or [Decimal.to_string/2](https://hexdocs.pm/decimal/Decimal.html#to_string/2)) for example.

The highest priority for `ex_money` is to preserve the integrity of a money-with-currency type and therefore I have no plan to modify `Money.to_string/2` from its current behaviour.

**Comment by kipcole9** [comment]
Daniel, glad you found a good way forward.  I know that there needs to be a better way to format and enter currency amounts in HTML and its on my "to do" list.  Thanks for letting me know what you did so I can share that if asked again.

**Comment by nicolasdabreo** [comment]
@kipcole9 Is this still the best solution for interfacing with forms? I'm struggling to get a working version using Money.

![Image](https://github.com/user-attachments/assets/e38d030e-c500-4cd9-94fd-1d44e658ba36)
![Image](https://github.com/user-attachments/assets/09caa994-71e7-48f3-b51e-fdee632640a3)

**Comment by kipcole9** [comment]
@nicolasdabreo, you're right, a proper component for money would be great. I don't know that I'm the right person to do that since I haven't done any liveview or modern phoenix.

I can however make it easier for you to output a formatted currency amount that doesn't have the currency symbol - on the assumption you're setting the currency itself in another field (which looks to be the case in your form).

In fact the capability is there - but you've just made me realise that (a) its not documented and (b) its not exposed in `ex_money`.

For immediate use you can do something like:
```elixir
iex> my_money = Money.new(:EUR, 1234.56)
iex> Cldr.Number.to_string my_money.amount, format: :currency_no_symbol, locale: :fr
{:ok, "1 234,56"}
```

I will now:
(a) Update the documentation for `Cldr.Number.to_string/2` to show the option `:currency_no_symbol` and 
(b) update `Money.to_string/2` to add the option `currency_symbol: :none`

Basically this means when I fix this (in the next 8 hours) you should find `Money.to_string(my_money, currency_symbol: :none)` useful for your form design.

Again, this is far from my area of strength so please do feel free to correct, guide and advise as you see fit.

**Comment by kipcole9** [comment]
I will also look in to the parse error you are seeing for the second example you show. I'm a bit ashamed to say I don't know what that script is. Can you let me know the locale? I'm not sure if its a bug or not but I'll look into it.

**Comment by nicolasdabreo** [comment]
That sounds great, thanks for looking into this for me. The locale is `ar-MA` and the symbol is the Morrorcan Dhiram.

**Comment by kipcole9** [comment]
> The locale is ar-MA and the symbol is the Morrorcan Dhiram.

Thanks, that helps. The parse error is most likely because I'm not pre filtering the [right-to-left mark](https://codepoints.net/U+200F?lang=en) so that should be an easy fix and I'll add a test. I also need to check if I'm prefiltering the [non-breaking space](https://en.wikipedia.org/wiki/Non-breaking_space).

**Comment by kipcole9** [comment]
I've published updated to `ex_cldr_currencies`, `ex_cldr_numbers` and `ex_money` that address these two issues.  You can now do the following to format a money without a currency symbol. I think that addresses your primary requirement.
```elixir
iex> m = Money.new "1234.56", :MAD
Money.new(:MAD, "1234.56")
iex> Money.to_string m, currency_symbol: :none, locale: "ar-MA"
{:ok, "1.234,56"}
```
I have also fixed the parsing of strings that have trailing RTL markers in them. For example:
```elixir
iex> {:ok, string} = Money.to_string(m, locale: "ar-MA")
{:ok, "\u200F1.234,56\u00A0د.م.\u200F"}
iex> Money.parse(string, locale: "ar-MA")
Money.new(:MAD, "1234.56")
```

Note that you will need to `mix deps.update ex_cldr_currencies ex_cldr_numbers ex_money` in order for these changes to apply.

Let me know if this solves your issue?
